src/nlexpr.jl

```diff
@@ -572,6 +572,22 @@ function expr_arrangeargs(args::Array; kwargs...)
     return
 end
 
+"""
+Check if it is an sample of `+()` (https://github.com/lanl-ansi/Alpine.jl/issues/221)
+"""
+function expr_is_emptysum(expr)
+    try
+        _ = eval(expr)
```

Comment:
Using `eval` is a strong signal that you're doing something wrong 😆 

If I understand correctly, all you need is:
```julia
expr_is_empty_sum(expr) = expr == :(+())
```

Reply:
Thanks, @odow!

I'm not very familiar with Julia's expressions 😄

Reply:
@odow BTW, why JuMP is adding this empty expression?

Reply:
I'm not sure. I'll take a look:
```julia
julia> model = Model();

julia> @variable(model, x);

julia> @expression(model, expr, x^2);

julia> @NLconstraint(model, expr >= 3)
(+() + x * x) - 3.0 ≥ 0
```